schema-manager/src/test/java/io/camunda/search/schema/SchemaManagerStartupIT.java

```diff
@@ -0,0 +1,117 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategyTarget;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+class SchemaManagerStartupIT {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SchemaManagerStartupIT.class);
+
+  private static final String CAMUNDA_TEST_IMAGE_NAME =
+      Optional.ofNullable(System.getenv("CAMUNDA_TEST_DOCKER_IMAGE"))
+          .orElse("camunda/camunda:SNAPSHOT");
+
+  private final AtomicBoolean isShutdown = new AtomicBoolean(false);
+
+  @AutoClose
+  private final ZeebeContainer camunda =
+      new ZeebeContainer(DockerImageName.parse(CAMUNDA_TEST_IMAGE_NAME))
+          .withEnv("CAMUNDA_DATABASE_URL", "http://test-elasticsearch:9200")
+          .withEnv("SPRING_PROFILES_ACTIVE", "broker")
+          .withNetwork(Network.SHARED)
+          .waitingFor(
+              new WaitStrategy() {
+                @Override
+                public void waitUntilReady(final WaitStrategyTarget waitStrategyTarget) {
+                  // Wait until the application is shutdown
+                  Awaitility.await()
+                      .timeout(Duration.ofSeconds(60))
+                      .pollDelay(Duration.ofSeconds(1))
+                      .untilTrue(isShutdown);
+                }
+
+                @Override
+                public WaitStrategy withStartupTimeout(final Duration startupTimeout) {
+                  return this;
+                }
+              });
+
+  @Container
+  private final ElasticsearchContainer es =
+      createDefeaultElasticsearchContainer()
+          // Enable security in ES for to trigger the retry logic as application does not have
+          // credentials
+          .withEnv("xpack.security.enabled", "true")
+          .withNetwork(Network.SHARED)
+          .withNetworkAliases("test-elasticsearch");
+
+  @AutoClose private final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  // false is to test the case when the shutdown is before the schema startup is started
+  // true is to test the case when the shutdown is when the schema startup is retrying
+  void shouldGracefullyShutdownWhenSchemaStartupStillRunning(final boolean waitForSchemaStartup)
+      throws InterruptedException {
+    // given
+    executor.submit(() -> camunda.start());
```

Comment:
:wrench: To avoid the overhead of managing an executor, why not just run it in a virtual thread?

e.g. 

```java
final var t = Thread.ofVirtual().start(() -> camunda.start());

// later when you want to join until it's gone (if you really need to?)
t.join(Duration.ofMinutes(1));
```

Then you don't need to keep track of an executor resource :shrug: